shipyard_blueprints/azure-data-factory/shipyard_azure_data_factory/azure_data_factory.py

```diff
@@ -0,0 +1,96 @@
+import time
+
+from azure.identity import ClientSecretCredential
+from azure.mgmt.datafactory import DataFactoryManagementClient
+from shipyard_templates import Etl, ShipyardLogger, ExitCodeException
+
+logger = ShipyardLogger.get_logger()
+
+
+class AzureDataFactoryClient(Etl):
+    def __init__(self, client_id, client_secret, tenant_id, subscription_id):
+        """
+        Initializes the AzureDataFactory instance with provided Azure credentials.
+
+        :param client_id: Azure client ID
+        :param client_secret: Azure client secret
+        :param tenant_id: Azure tenant ID
+        :param subscription_id: Azure subscription ID
+        """
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.tenant_id = tenant_id
+        self.subscription_id = subscription_id
+        self.adf_client = None
+
+    def setup_client(self):
```

Comment:
Since you are referencing the `self.adf_client` in the `trigger_sync` method, and doing a check that it exists, I would recommend setting a property for it that calls this setup_client. Something like: 

```python
def __init__(...):
    ... 
    self._adf_client = None

@property 
def adf_client(self):
    if not self.adf_client:
        self._adf_client = self.setup()
    return self._adf_client
```
That was the approach I used in the databases and it worked pretty well

Reply:
Agreed- I think this should be the standard. So push me again if I ever forget again